staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -520,6 +520,25 @@ type PersistentVolumeClaimSpec struct {
 	// (Beta) Using this field requires the AnyVolumeDataSource feature gate to be enabled.
 	// +optional
 	DataSourceRef *TypedLocalObjectReference `json:"dataSourceRef,omitempty" protobuf:"bytes,8,opt,name=dataSourceRef"`
+	// (Alpha) Using this field requires the CrossNamespaceSnapshots feature gate to be enabled.
```

Comment:
We'll want more description here similar to DataSourceRef. And also mention that only one of DataSourceRef and DataSourceRef2 can be specified.

Also following the guidance here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#object-references, we'll want to be more specific on what happens when the object doens't exist, and also what happens to the reference after the PVC is created.